### ISSUE TITLE
About the dataset

### ISSUE THREAD

**Issue by slacklife** [yl-1993/learn-to-cluster#63]
Hi @yl-1993, really an amazing work!
I have some questions about the dataset.
In https://github.com/yl-1993/learn-to-cluster/blob/master/DATASET.md
1. Is the feature of [ms1m_part1.tar.gz](https://drive.google.com/open?id=16WD4orcF9dqjNPLzST2U3maDh2cpzxAY)
extracted with the [resnet50_part0_train.pth.tar](https://drive.google.com/open?id=1eKsh7x-RUIHhIJ1R9AlUjsJdsdbh2qim) ?
2. What is the training data of the model [resnet50_part0_train.pth.tar](https://drive.google.com/open?id=1eKsh7x-RUIHhIJ1R9AlUjsJdsdbh2qim). Are they the Images of part0_train.bin?
3. If the pretrained model is trained with the images of part0_train.bin, and then train clustering on the part1_train.bin, What if I use a model to extrace feature which is trained by other images set and then use that feature to train clustering?

Thanks!

**Comment by yl-1993** [comment]
Hi @slacklife , thanks for checking out our work.

1. Yes, all the features, from part0 to part9, are extracted with the same model trained on part0.
2. The training data is part0. We first train on part0 to get a model and then use the model to extract features for part0.
3. Yes. For our experiments, we want to re-use the annotated data, and thus we use the labeled data to train both feature extractor and clustering model. It indicates that we can discover more knowledge from the same amount of training data. And I think it will get better results by using separated data for training feature extractor and clustering model. Using separated data will help to reduce the statistical gap of edge similarities between training and testing, otherwise the edge similarities on training graphs is much higher than testing graphs. (Actually I have received some emails about trying similar ideas and they seemed to work well.)

**Comment by slacklife** [comment]
Hi, many thanks for your reply ! 
I still have another question want to figure out.

HFsoftmax is designed for massive classification and HFsoftmax's result is slightly lower than full softmax in table1 of [paper](https://arxiv.org/pdf/1801.01687.pdf). 
There are only about 8.6K identities in part0, not as many as 100K identities in full Celeb-1M and 672K identities in MegaFace. Why do you use hfsoftmax to train [resnet50_part0_train.pth.tar](https://drive.google.com/open?id=1eKsh7x-RUIHhIJ1R9AlUjsJdsdbh2qim) instead of the full softmax or cos face(I found there is the cos face classifier implementation in [hfsoftmax](https://github.com/yl-1993/hfsoftmax/blob/80fa7d333b/models/classifier.py#L28)) ? These method maybe get a better result.

**Comment by yl-1993** [comment]
@slacklife You are right. Actually, to avoid interference from irrelevant factors, we use the *full softmax* for training instead of hfsoftmax and CosFace. To be more clear, we mentioned hfsoftmax here only to use the shared code for face recognition.

Another paper from our group, [CDP](http://mmlab.ie.cuhk.edu.hk/projects/CDP/), has studied the influence of models trained with different losses. It shows that better initial model often lead to better clustering results.

**Comment by houguanqun** [comment]
Hi, @slacklife Could you unzip the file  “[resnet50_part0_train.pth.tar](https://drive.google.com/file/d/1eKsh7x-RUIHhIJ1R9AlUjsJdsdbh2qim/view)”

**Comment by yl-1993** [comment]
Hi @houguanqun, please refer to the discussion of pretrained model in https://github.com/yl-1993/learn-to-cluster/issues/75.
